static/js/shared/source_selector.tsx

```diff
@@ -0,0 +1,294 @@
+      if (!importName) {
+        sourceOptionsNoImportName.push(metahash);
+        return;
+      }
+      if (!(importName in importNameToSourceOptions)) {
+        importNameToSourceOptions[importName] = [];
+      }
+      importNameToSourceOptions[importName].push(metahash);
+      if (importNameToSourceOptions[importName].length > 3) {
+        shouldShowSections = true;
+      }
+    });
+    if (shouldShowSections) {
+      const sortedImportNames = Object.keys(importNameToSourceOptions).sort();
+      return (
+        <>
+          {sortedImportNames.map((importName) => (
+            <Collapsible
+              key={svInfo.dcid + importName}
+              trigger={getImportTriggerJsx(false, importName)}
+              triggerWhenOpen={getImportTriggerJsx(true, importName)}
+            >
+              {importNameToSourceOptions[importName].map((metahash) =>
+                getSourceOptionJsx(svInfo, metahash)
+              )}
+            </Collapsible>
+          ))}
+          {sourceOptionsNoImportName.map(
+            (metahash) =>
+              metahash in svInfo.metadataMap &&
+              getSourceOptionJsx(svInfo, metahash)
+          )}
+        </>
+      );
+    } else {
+      return (
+        <>
+          {Object.keys(svInfo.metadataMap).map((metahash) =>
+            getSourceOptionJsx(svInfo, metahash)
+          )}
+        </>
+      );
+    }
+  }
+
+  function onConfirm(): void {
+    props.onSvMetahashUpdated(modalSelections);
+    setModalOpen(false);
+  }
+}
+
+function getSVTriggerJsx(
+  opened: boolean,
+  svName: string,
+  selectedMetadata: StatMetadata
+): JSX.Element {
+  const sourceTitle = getSourceTitle(selectedMetadata);
+  return (
+    <div
+      className={
+        opened
+          ? `${SELECTOR_PREFIX}-trigger ${SELECTOR_PREFIX}-sv-trigger-opened`
+          : `${SELECTOR_PREFIX}-trigger ${SELECTOR_PREFIX}-sv-trigger-closed`
+      }
+    >
+      <div className={`${SELECTOR_PREFIX}-trigger-title`}>
+        {svName}
+        <div className={`${SELECTOR_PREFIX}-trigger-byline`}>
+          source: {sourceTitle}
+        </div>
+      </div>
+      {opened ? UP_ARROW_HTML : DOWN_ARROW_HTML}
+    </div>
+  );
+}
+
+function getImportTriggerJsx(opened: boolean, title: string): JSX.Element {
+  return (
+    <div
+      className={
+        opened
+          ? `${SELECTOR_PREFIX}-trigger ${SELECTOR_PREFIX}-import-trigger-opened`
+          : `${SELECTOR_PREFIX}-trigger ${SELECTOR_PREFIX}-import-trigger-closed`
+      }
+    >
+      {opened ? MINUS_HTML : PLUS_HTML}
+      <div className={`${SELECTOR_PREFIX}-trigger-title`}>{title}</div>
+    </div>
+  );
+}
+
+function getModalSelections(
+  svInfo: SourceSelectorSvInfo[]
+): Record<string, string> {
+  const result = {};
+  svInfo.forEach((info) => (result[info.dcid] = info.metahash));
+  return result;
+}
+
+function getSourceTitle(metadata: StatMetadata): string {
+  if (_.isEmpty(metadata)) {
+    return "Best Available";
```

Comment:
We should add note somewhere on what is "Best Available"..

Reply:
done